### PR TITLE
Refactor code for fetching active tracks using new API

### PR DESCRIPTION
**Pull request by abhishekbose87**
Issue #2633

### CODE REVIEW

**Comment by abhishekbose87** [comment]
@kytrinyx Do you think this is a good way to go about this change ? Also need to figure out how to invalidate this cache when there is any change in active tracks

File: x/track.rb [hunk 1]
```diff
@@ -8,6 +10,12 @@ def self.all
       end
     end
 
+    def self.active
```
Comment (kytrinyx) [inline x/track.rb]: I'd rather not have this method here. The `X::Track` class feels like it should be as generic as possible.

**Comment by kytrinyx** [comment]
I think we should cache up in the route. I think it would be fine to cache the entire object rather than pulling out just the id and the name in a hash. I much prefer to be sending messages to objects than indexing into a hash.

**Comment by abhishekbose87** [comment]
@kytrinyx I have made some changes to my solution. Can you please review it now.

One problem that I see is that we don't have the same set of attributes from v3 API as we have in the previous version. So I had to do a workaround to get slug. 
```
alias_method :slug, :id
```
And I could not replace the `tracks` method inside `routes/core.rb`to use new API, as `upcoming` attribute is not being fetched which causes `/about` page to fail.

**Comment by kytrinyx** [comment]
Ah, I'm going to rewrite the about page. Upcoming is inactive tracks with more than 1 exercise on it.

**Comment by kytrinyx** [comment]
I'll review this today, thanks.

File: app/routes/core.rb [hunk 2]
```diff
@@ -125,6 +125,10 @@ def tracks
           ExercismWeb::Presenters::Tracks.tracks
         end
 
+        def active_tracks
+          X::Track.all.select(&:active)
```
Comment (kytrinyx) [inline app/routes/core.rb]: let's cache this with  `@active_tracks ||=`

At the moment I'm going to have to restart the app anyway if we activate new languages. I'll think about how to deal with that differently when I start tackling overall performance stuff.

Comment (abhishekbose87) [inline app/routes/core.rb]: Okies :+1:

**Comment by abhishekbose87** [comment]
Thanks for your inputs. Let me know if this good enough to be merged or there are other changes required.

File: app/routes/core.rb (hunk 2 above)
Comment (abhishekbose87) [inline app/routes/core.rb]: I have a stupid question here. By using `@active_tracks ||=` are we caching the data across many requests or a single request. I tested and found that it sends a API request for every refresh. Am I doing something wrong ?

**Comment by kytrinyx** [comment]
This looks good, thank you!

**Comment by abhishekbose87** [comment]
@kytrinyx I have a question here. By using @active_tracks ||= are we caching the data across many requests or a single request. I tested and found that it sends a API request for every refresh. Is this the expected behaviour ?

**Comment by kytrinyx** [comment]
I believe it's being cached for a single request. If we want to add caching across multiple requests we'd need to put it in a class-level variable.

**Comment by abhishekbose87** [comment]
Ok .. got it. Thanks :smile:

**Comment by kytrinyx** [comment]
I fixed it yesterday :)

**Comment by abhishekbose87** [comment]
:+1:

**Pull request merged**